fvm/meter/meter.go

```diff
@@ -557,3 +573,23 @@ func GetStorageKeyValueSizeForTesting(
 func (m *Meter) GetStorageUpdateSizeMapForTesting() MeteredStorageInteractionMap {
 	return m.storageUpdateSizeMap
 }
+
+func (m *Meter) MeterEmittedEvent(byteSize uint64) error {
+	m.totalEventEmitBytes += uint64(byteSize)
+	m.eventCounter++
+
+	if m.totalEventEmitBytes > m.eventEmitByteLimit {
```

Comment:
in current use case the `enforceLimit` is handled at `fvm/environment/meter.go`, so I'm not adding an extra flag here.